# TechnoHacks EduTech Internship

# Data Analytics Tasks : By Vivek Kumar

# Task 5 - Remove Duplicates : Identify and remove duplicate values in a dataset.

# Import Libraries

Importing the necessary libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


# Load the Iris dataset

In [2]:
iris_data=pd.read_csv('iris.csv',encoding='utf8')
data=iris_data.copy()

Show all data. which is present in iris.csv file

In [3]:
data

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
0      1            5.1           3.5            1.4           0.2   
1      2            4.9           3.0            1.4           0.2   
2      3            4.7           3.2            1.3           0.2   
3      4            4.6           3.1            1.5           0.2   
4      5            5.0           3.6            1.4           0.2   
..   ...            ...           ...            ...           ...   
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
0       Iris-setosa  
1       Iris-setosa  
2       Iris-setosa  
3       Iris-setosa  
4       Iris-setosa  
..              ...  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica  

[150 rows x 6 columns]

Before removing duplicate value shape and size of data

In [4]:
data.shape

(150, 6)

In [5]:
data.size

900

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB


In [7]:
# print first five data 
data.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [8]:
# print last five data 
data.tail()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica

In [9]:
# describe the data
data.describe()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000

In [10]:
# first of check null value.
data.isnull().sum()

Id               0
SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

In [11]:
# find the data types
data.dtypes

Id                 int64
SepalLengthCm    float64
SepalWidthCm     float64
PetalLengthCm    float64
PetalWidthCm     float64
Species           object
dtype: object

# Identify Duplicates

The most important method to deal with duplicates is duplicated method which will tell which values are duplicate.

In [12]:
# identifying duplicate row
data.duplicated()           # no any duplicate data present in rows

0      False
1      False
2      False
3      False
4      False
       ...  
145    False
146    False
147    False
148    False
149    False
Length: 150, dtype: bool

In [18]:
# identifying duplicate columnwise
data.duplicated(data.columns,keep='first')

0      False
1      False
2      False
3      False
4      False
       ...  
145    False
146    False
147    False
148    False
149    False
Length: 150, dtype: bool

In [19]:
# find the sum of all duplicate value
data.duplicated(data.columns,keep='first').sum()

0

In [20]:
#finding duplicate value after compare of these four column
data.duplicated(subset=['SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],keep='first')

0      False
1      False
2      False
3      False
4      False
       ...  
145    False
146    False
147    False
148    False
149    False
Length: 150, dtype: bool

In [22]:
# it basically find how many row are duplicate
data.duplicated(subset=['SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],keep='first').sum()

3

# Removing Duplicates

Using drop_duplicates method to get rid of duplicate values.

In [23]:
data.drop_duplicates(subset=['SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],keep='first',inplace=True)

Check whether duplicates are removed

In [24]:
data.duplicated(subset=['SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'],keep='first').sum()

0

# After removing the duplicate value check the shape of data

In [25]:
data.shape

(147, 6)

In [26]:
data.size

882

Observation: Before removing duplicates from the dataset the shape was (150, 6) i.e there were 150 rows and 6 columns. After removing duplicates the shape is (147, 6) i.e there are 147 rows and 6 columns. Hence, it is observed that there were 3 duplicated rows in the Iris dataset which are successfully removed.

Thus, Task 3 for identifying and removing duplicate values in a dataset has been completed.